llvm/lib/CodeGen/MachineTraceMetrics.cpp

```diff
@@ -852,14 +846,14 @@ computeInstrDepths(const MachineBasicBlock *MBB) {
   // implies Head->HasValidInstrDepths, so we only need to start from the first
   // block in the trace that needs to be recomputed.
   SmallVector<const MachineBasicBlock*, 8> Stack;
-  do {
+  while (MBB) {
```

Comment:
Is this one NFC though?

Reply:
I agree this code is better, but there is no assertion that `MBB != nullptr`.

Reply:
The MBB->getNumber() in the loop would fail on MBB = nullptr.

Reply:
So then it sounds like this patch contains a potential bug fix, which is not NFC?

Reply:
Yes, technically. I've dropped the NFC tag.